test/system/StringTest.ooc

```diff
@@ -149,6 +192,114 @@ StringTest: class extends Fixture {
 			expect(decdabbc, is equal to("decdabbc"))
 			(bc, cd, decdabbc) free()
 		})
+		this add("reverse", func {
+			This reverseTest("", "")
+			This reverseTest(" ", " ")
+			This reverseTest("1", "1")
+			This reverseTest("12", "21")
+			This reverseTest("a", "a")
+			This reverseTest("ab", "ba")
+			This reverseTest("abc", "cba")
+			This reverseTest("Test", "tseT")
+			This reverseTest("Test!", "!tseT")
+			This reverseTest("marcus", "sucram")
+			This reverseTest("sucram", "marcus")
+			This reverseTest("Testing", "gnitseT")
+			This reverseTest("Reversing a very long sentence just in case of weird upper limits when allocating the result.", ".tluser eht gnitacolla nehw stimil reppu driew fo esac ni tsuj ecnetnes gnol yrev a gnisreveR")
+		})
+		this add("findAll", func {
+			This findTest("", "")
+			This findTest("a", "")
+			This findTest("", "a")
+			This findTest("abab", "ab", 0, 2)
+			This findTest("abcab", "ab", 0, 3)
+			This findTest("ababa", "aba", 0) // Finding following overlaps would make it useless for split
```

Comment:
Sure, I'm not saying there's anything wrong with it, especially for `split` or `replace`, but there are instances where overlaps are *also* acceptable.